diff --git a/node_modules/@ant-design/pro-table/es/Table.js b/node_modules/@ant-design/pro-table/es/Table.js
index 850c462..ffa210f 100644
--- a/node_modules/@ant-design/pro-table/es/Table.js
+++ b/node_modules/@ant-design/pro-table/es/Table.js
@@ -740,7 +740,7 @@ var ProTable = function ProTable(props) {
       var _preserveRecordsRef$c;
       return (_preserveRecordsRef$c = preserveRecordsRef.current) === null || _preserveRecordsRef$c === void 0 ? void 0 : _preserveRecordsRef$c.get(key);
     });
-  }, [selectedRowKeys]);
+  }, [action.dataSource, selectedRowKeys]);
 
   /** 内置的工具栏 */
   var toolbarDom = toolBarRender === false ? null : /*#__PURE__*/_jsx(Toolbar, {
diff --git a/node_modules/@ant-design/pro-table/lib/Table.js b/node_modules/@ant-design/pro-table/lib/Table.js
index 9787e2e..5a7f607 100644
--- a/node_modules/@ant-design/pro-table/lib/Table.js
+++ b/node_modules/@ant-design/pro-table/lib/Table.js
@@ -743,7 +743,7 @@ var ProTable = function ProTable(props) {
       var _preserveRecordsRef$c;
       return (_preserveRecordsRef$c = preserveRecordsRef.current) === null || _preserveRecordsRef$c === void 0 ? void 0 : _preserveRecordsRef$c.get(key);
     });
-  }, [selectedRowKeys]);
+  }, [action.dataSource, selectedRowKeys]);
 
   /** 内置的工具栏 */
   var toolbarDom = toolBarRender === false ? null : /*#__PURE__*/(0, _jsxRuntime.jsx)(_ToolBar.default, {
